apis/python/src/tiledbsoma/_dense_nd_array.py

```diff
@@ -120,13 +123,26 @@ def read(
         #
         # The only exception is if the array has been created but no data have been written at
         # all, in which case the best we can do is use the schema shape.
-        data_shape = self._handle.schema.shape
+        handle: clib.DenseNDArrayWrapper = self._handle._handle
+        data_shape = handle.shape
         ned = self.non_empty_domain()
         if ned is not None:
             data_shape = tuple(slot[1] + 1 for slot in ned)
         target_shape = dense_indices_to_shape(coords, data_shape, result_order)
 
-        sr = self._soma_reader(result_order=result_order)
+        to_clib_result_order = {
+            options.ResultOrder.AUTO: clib.ResultOrder.automatic,
+            options.ResultOrder.ROW_MAJOR: clib.ResultOrder.rowmajor,
+            options.ResultOrder.COLUMN_MAJOR: clib.ResultOrder.colmajor,
+            "auto": clib.ResultOrder.automatic,
+            "row-major": clib.ResultOrder.rowmajor,
+            "column-major": clib.ResultOrder.colmajor,
```

Comment:
you can avoid this by writing `result_order = options.ResultOrder(result_order)` above, which will handle the enumification directly:

```
>>> import enum
>>> class X(enum.Enum):
...     ONE = 1
...     TWO = 2
...     THREE = 3
...     
... 
>>> X(1)
<X.ONE: 1>
>>> X(X.ONE)
<X.ONE: 1>
```

This would also obviate the `if not in` below.